DetectDynamicJS.py

```diff
@@ -71,28 +72,46 @@ def doPassiveScan(self, baseRequestResponse):
         # This is, because the insertionPoint idea doesn't work well
         # for this test.
         scan_issues = []
-
         if not self.isGet(baseRequestResponse.getRequest()):
             baseRequestResponse = self.switchMethod(baseRequestResponse)
         if (not self.isScannableRequest(baseRequestResponse) or
             not self.isScript(baseRequestResponse) or
             self.isProtected(baseRequestResponse)):
             return None
         newRequestResponse = self.sendUnauthenticatedRequest(baseRequestResponse)
-        issue = self.compareResponses(newRequestResponse, baseRequestResponse)
-        if not issue:
-            return None
-        # If response is script, check if script is dynamic
-        if self.isScript(newRequestResponse):
-            # sleep, in case this is a generically time stamped script
-            sleep(1)
-            secondRequestResponse = self.sendUnauthenticatedRequest(baseRequestResponse)
-            isDynamic = self.compareResponses(secondRequestResponse, newRequestResponse)
-            if isDynamic:
-                issue = self.reportDynamicOnly(newRequestResponse, baseRequestResponse,
-                                               secondRequestResponse)
-        scan_issues.append(issue)
-        return scan_issues
+        if(self.isScannableRequest(newRequestResponse)):
+            issue = self.compareResponses(newRequestResponse, baseRequestResponse)
+            if not issue:
+                return None
+            # If response is script, check if script is dynamic
+            if self.isScript(newRequestResponse):
+                # sleep, in case this is a generically time stamped script
+                sleep(1)
+                secondRequestResponse = self.sendUnauthenticatedRequest(baseRequestResponse)
+                isDynamic = self.compareResponses(secondRequestResponse, newRequestResponse)
+                if isDynamic:
+                    issue = self.reportDynamicOnly(newRequestResponse, baseRequestResponse,
+                                                   secondRequestResponse)
+            scan_issues.append(issue)
+            return scan_issues
+        else:
```

Comment:
It looks to me like this else is not necessary. safe an indent. safe a little cats life. Makes the code more readable.

Reply:
This is fixed as per comments.